src/v/http/client.h

```diff
@@ -360,8 +360,16 @@ redacted_fields();
 ss::future<boost::beast::http::status>
 status(client::response_stream_ref response);
 
+template<typename T = iobuf>
+requires std::same_as<T, iobuf> || std::same_as<T, void>
+ss::future<T> drain(client::response_stream_ref response);
+
+template<>
 ss::future<iobuf> drain(client::response_stream_ref response);
 
+template<>
+ss::future<> drain(client::response_stream_ref response);
```

Comment:
maybe `drain` and `drain_to_iobuf`? is the template stuff so that you can have a function with the same signature differing only by the return value?

Reply:
Looks neat with a template :)

Reply:
> Let's experiment with leaving it like this and see what experience will tell.

the outcome of the experiment is already known: we'll use the templated version because the right time to change it was before merging and now its not worth changing.